config/pmd.xml

```diff
@@ -159,6 +202,22 @@
     <!--we do not care about this minor overhead, we are not Android application and we do not want to change
      visibility of methods. Package-private visibility should be avoid almost always.-->
     <exclude name="AccessorMethodGeneration"/>
+    <!-- Conflicts with the ToArrayCallWithZeroLengthArrayArgument check from the Idea
+         This rule is based on a study by Aleksey Shipilёv
+         https://shipilev.net/blog/2016/arrays-wisdom-ancients/
+         However, in modern JVM the result is different:
+         Benchmark (size)  (type)   Mode  Cnt Score    Error  Units
+         simple     1000  arraylist avgt  15  400.156   4.154 ns/op
+         sized      1000  arraylist avgt  15 1051.462  26.820 ns/op
+         zero       1000  arraylist avgt  15  743.794  27.400 ns/op
+         simple     1000  hashset   avgt  15 4728.179 130.822 ns/op
+         sized      1000  hashset   avgt  15 4960.655 179.637 ns/op
+         zero       1000  hashset   avgt  15 5101.816 159.180 ns/op
+         The advantages of this rule are questionable, and the flaws are obvious.
+     -->
+    <exclude name="OptimizableToArrayCall"/>
+    <!-- It produces too much violations, suppressed till we figure out how useful that metrics -->
+    <exclude name="DataClass"/>
```

Comment:
change comment to:
"we do not follow philosophy of complete encapsulation, we like data classes(preferable immutable) to transfer content from one part to another. Too much "false-positives" on Checks classes. No way to customize rule(no properties)."

Reply:
done